server/message_hooks_test.go

```diff
@@ -822,7 +821,299 @@ func TestReactionHasBeenRemoved(t *testing.T) {
 	})
 }
 
+func TestMessageHasBeenPosted(t *testing.T) {
```

Comment:
This is the bulk of this change -- most other diffs are just files being split apart for clarity.